plugins/modules/gitlab_runner.py

```diff
@@ -216,6 +216,7 @@ def create_or_update_runner(self, description, options):
                 'locked': options['locked'],
                 'run_untagged': options['run_untagged'],
                 'maximum_timeout': options['maximum_timeout'],
+                'access_level': options['access_level'],
```

Comment:
This looks ok to me, but I think it has a slight change in runner registration behavior: this module defines `ref_protected` as the default for creation, which I think is not the real default in GitLab's API if the attribute is not provided*.
https://github.com/ansible-collections/community.general/blob/13b9d949fc20378c42d8c5dcbfe3034c1894ddeb/plugins/modules/gitlab_runner.py#L331

But I guess this happens anyway after the first run with the current approach, since the update method will then change it, and that is the reason you opened the issue in the first place I assume. 

Just wanted to point this out but will let @felixfontein decide on the best approach.

*See https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner

Reply:
A little side note: Not sure about the history behind this, but I think selecting `ref_protected` as the default here was a bit of an exotic choice. Of course that's more secure as it limits the runner to being available on protected branches only, but I'm not sure that's the more common use case for GitLab runners.

Reply:
That is good remark. I was not aware of that fact. From that point of view, it would be the best to also change the default value of `access_level` to `not_protected`. Because the `access_level` was not passed to the runner creation call before, people actually got an unprotected runner. That is not what you would expect from the documentation. But it is inline with the default behavior of the GitLab API.

Reply:
@cfiehe I agree that makes more sense and I see you've already incorporated it. The only thing now is that this is technically a breaking change so I'm eager to hear feedback from the collection maintainers (I'm only involved in the GitLab module here :))

Reply:
Changing the default value is a breaking change and cannot be merged (see also #582). Changing the default requires a longer deprecation period.

Generally I would avoid having a default *at all* though, then the default on creation would be "use whatever GitLab thinks is the default", and the default for updating would be "don't update that value". Whether you prefer to force a specific value for this option if none is provided, or whether not specifying a value means "I don't care / don't change", that's something for the module maintainers to decide. But it's nothing this bugfix PR should touch/change.

Reply:
@felixfontein the issue is that the bugfix, without removing the default, would also be a breaking change.

So maybe removing the default here is the easiest transition, but then the code will need to ensure that it's not sending null values to the API if no attribute is present, as GitLab sometimes doesn't like that.

Reply:
That's true, but at least it's restoring the documented behavior, and not changing the documented behavior *and* the implementation.

But yeah, it's also not helping that the current behavior has been there for quite a few years (it has always been like this since it was migrated to this repository). That's also (in part) why I didn't add the `backport-5` label.

So maybe the best way forward is:
1. Add a new option `access_level_on_creation` of type boolean with no explicit default, but implicit default `false`.
2. If that option is not specified, a deprecation warning will be triggered that this option will switch to `true` for the next major release (community.general 7.0.0).
3. Only set the access level on creation when `access_level_on_creation=true` is passed to the module.

It would be a rather short deprecation period, but at least users would know that something is changing, and (most importantly) can get the module to use the new behavior already now. What do you think about this approach?

(Deprecating the default value of `access_level` is something that should probably happen in a separate PR, as it's strictly not related to this problem.)